context/app/static/js/components/workspaces/Tables/WorkspaceItemsTable/hooks.ts

```diff
@@ -15,7 +15,11 @@ function useWorkspaceItemsTable<T extends WorkspaceItem>({
   toggleAllItems,
 }: WorkspaceItemsTableProps<T>) {
   const [sortField, setSortField] = useState<SortField>(initialSortField);
-  const [numVisibleItems, setNumVisibleItems] = useState(showSeeMoreOption ? 3 : items.length);
+  const [numVisibleItems, setNumVisibleItems] = useState(3);
+
+  useEffect(() => {
+    setNumVisibleItems(showSeeMoreOption ? 3 : items.length);
+  }, [items, showSeeMoreOption]);
```

Comment:
If this is the only place where we use the `setNumVisibleItems` setter and it is fully derived from other variables, we can avoid making this a stateful variable altogether. A quick `ctrl + F` didn't find any usage of this function outside of this hook, so I think this should work and avoid the extra rerender.

```suggestion
  const numVisibleItems = showSeeMoreOption ? 3 : items.length;
```

Reply:
Unfortunately this setter is in use by the "See more" button (in order to increase the number of visible rows by three at a time).